javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -783,7 +824,51 @@ public boolean isTraversableAgain() {
 
     @Override
     public Iterator<T> iterator() {
-        return isEmpty() ? Iterator.empty() : trie.iterator();
+        return isEmpty() ? Iterator.empty()
+                         : new VectorIterator();
+    }
+
```

Comment:
> Thanks, but I did test it and it was surprisingly slow.

Really?

There might be a little overhead for creating 4 Iterators instead of 1 but this is what I expect:

For Iterating a small amount of elements the constant factor of creating 4 Iterators will be weighted much.

I.e. when Iteration takes 8 ms for one Iterator and 16 ms using 4 Iterators the _relative_ diff looks huge, but _really_ it isn't!

I.e. when Iteration takes 10008 ms for one Iterator and 100016 ms using 4 Iterators the _relative_ diff is near zero (+/- measure error).

---

Because the BMT.iterator() is nearly perfect and also the Object[] iterators are very efficient loops, the diff can't be huge. Our test cases are not right (interpreted).